### PR TITLE
Add interesting fields to usage report (fixes #559)

### LINKED ISSUE #559: data.syncthing.net request (new metric to gather) rescan interval
Just a random curiosity I have if people change the rescan interval from default.  Like a server I use that never adds anything and is just an offsite backup I have rescanning once an hour.  Not sure how to aggregate that now that you can change it independent of repo, probably just average from all repos. From programming aspect it would be good to see how changing that reduces the resource usage as well.

I know you have to reask for confirmation whenever you change that ( :+1: btw) so understand if this is on hold for a few months if you think of some more things to add.

### PR DESCRIPTION
**Pull request by calmh**
This adds a number of new fields to the anonymous usage report (and this bumps the version). The new fields look like this in the report and are mostly about feature usage:

```json
{
  "announce": {
    "defaultServersDNS": 2,
    "defaultServersIP": 0,
    "globalEnabled": true,
    "localEnabled": true,
    "otherServers": 0
  },
  "deviceUses": {
    "compressAlways": 0,
    "compressMetadata": 3,
    "compressNever": 0,
    "customCertName": 0,
    "dynamicAddr": 0,
    "introducer": 0,
    "staticAddr": 3
  },
  "folderUses": {
    "autoNormalize": 3,
    "ignoreDelete": 0,
    "ignorePerms": 0,
    "readonly": 0
  },
  "numCPU": 8,
  "rescanIntvs": [
    15,
    15,
    15
  ],
}
```

(Needs a companion change to `ursrv` to pick up these fields)

### CODE REVIEW

File: cmd/syncthing/usage_report.go [hunk 1]
```diff
@@ -120,6 +123,68 @@ func reportData(m *model.Model) map[string]interface{} {
...
+			res["usesReadonly"] = true
+		}
+		if cfg.IgnorePerms {
+			res["usesIgnorePerms"] = true
+		}
+		if cfg.IgnoreDelete {
+			res["usesIgnoreDelete"] = true
+		}
+		if cfg.AutoNormalize {
+			res["usesAutoNormalize"] = true
```
Comment (AudriusButkevicius) [inline cmd/syncthing/usage_report.go]: All 4 of these could be counts to give a better picture.

File: cmd/syncthing/usage_report.go [hunk 2]
```diff
@@ -120,6 +123,68 @@ func reportData(m *model.Model) map[string]interface{} {
...
+			res["usesCompressNever"] = true
+		}
+		for _, addr := range cfg.Addresses {
+			if addr == "dynamic" {
+				res["usesDynamicAddr"] = true
+			} else {
+				res["usesStaticAddr"] = true
+			}
+		}
+	}
```
Comment (AudriusButkevicius) [inline cmd/syncthing/usage_report.go]: Counts too

File: cmd/syncthing/usage_report.go [hunk 3]
```diff
@@ -120,6 +123,68 @@ func reportData(m *model.Model) map[string]interface{} {
...
+		if cfg.CertName != "" && cfg.CertName != "syncthing" {
+			res["usesCustomCertName"] = true
+		}
+		if cfg.Compression == protocol.CompressAlways {
+			res["usesCompressAlways"] = true
+		}
+		if cfg.Compression == protocol.CompressMetadata {
+			res["usesCompressMetadata"] = true
+		}
+		if cfg.Compression == protocol.CompressNever {
```
Comment (AudriusButkevicius) [inline cmd/syncthing/usage_report.go]: These could be elseifs

File: cmd/syncthing/usage_report.go (hunk 1 above)
Comment (AudriusButkevicius) [inline cmd/syncthing/usage_report.go]: Also, you can you an anno mous struct to avoid type assertions

Comment (calmh) [inline cmd/syncthing/usage_report.go]: slices of bool?

**Comment by AudriusButkevicius** [comment]
Do we know that the upgrade ur feature works? Perhaps we should explain that stuff has changed since we last prompted, and give the rationale why we are adding this (to see which features are much used and would benefit users if improved.l)

**Comment by calmh** [comment]
Counts fixed above. I haven't tested the ur upgrade thing lately; if it works (which I think it does), it just presents the normal dialog as if ur was disabled before. We could add a string somewhere about what changed and so on, but that then requires localization and crap...

**Comment by calmh** [comment]
Btw the new fields break the preview dialog on my browser - the fields are too many the modal isn't scrollable as it should be. I'll see if I can get it scrollable, or maybe have a scrollable textarea for the report instead...

**Comment by calmh** [comment]
With scrollable textarea:

screen shot 2015-09-06 at 17 04 31

**Comment by calmh** [comment]
Can add some more wordage about why this is added in the release notes too, not that anyone reads them :imp:

**Comment by AudriusButkevicius** [comment]
Another interesting metric would be discovery server counts and counts which are equal to defaults, while we are at it?

**Comment by AudriusButkevicius** [comment]
Though that needs special handling for android dns issue.

**Comment by calmh** [comment]
Yeah, I'll add that. I was actually considering it, it just felt like too much work. ;)

**Comment by AudriusButkevicius** [comment]
We already report status, hence it shouldn't be rocket science.

**Comment by calmh** [comment]
Added. It's a little bit fragile to check for the IP as I do there, but I don't know what's better and it's not likely to change in the very near future.

**Comment by AudriusButkevicius** [comment]
Amazing stuff

**Comment by AudriusButkevicius** [comment]
Also, relay addresses would be interesting too I guess. Blah, later.

**Comment by calmh** [comment]
Yeah, should do the same there. Also I'm going to add the `urVersion` to simplify for the server side.

**Pull request merged**